### PR TITLE
Add logging, assertions, DG, PPP

### PR DESCRIPTION
**Pull request by rharwo**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -459,10 +459,12 @@ The income or spending will be deleted from its corresponding list using its ind
 
 ## Product scope
 ### Target user profile
-1. prefer desktop apps over other types
+1. Prefer desktop apps over other types
```
Comment (NigelYeoTW) [inline docs/DeveloperGuide.md:462]: other types seems a bit too general, maybe leave it out or be more specific?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -459,10 +459,12 @@ The income or spending will be deleted from its corresponding list using its ind
...
+2. Prefer typing
```
Comment (NigelYeoTW) [inline docs/DeveloperGuide.md:463]: maybe add the part on fast typing as well since it was part of the project requirements

**Review by NigelYeoTW — COMMENTED** [review]
overall lgtm, just some small suggestions

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (rharwo) [reply docs/DeveloperGuide.md]: Fixed!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (rharwo) [reply docs/DeveloperGuide.md]: Fixed!

**Comment by NigelYeoTW** [comment]
lgtm, thanks

**Pull request merged**